### ISSUE TITLE
security fix: pin or replace `colors` dependency

### ISSUE THREAD

**Issue by micalevisk** [foreversd/forever#1124]
https://github.com/foreversd/forever/blob/2211e32a288b97c3c3d1e27f41370a9a489ee833/package.json#L24

`colors` was intentionally compromised by the author. The latest working version is `1.4.0`. So I believe you need to pin that version to `1.4.0` to prevent issues from the next upgrades.

![image](https://user-images.githubusercontent.com/13461315/148703991-ab465d30-52a6-4296-8d15-50e76617734d.png)

![image](https://user-images.githubusercontent.com/13461315/148704015-d08e1056-2fc9-41c8-9e86-7ae4a227fb90.png)

**Comment by kibertoad** [comment]
@micalevisk Could you please send a PR for that?

**Comment by micalevisk** [comment]
the version used here is `0.6.2`. Do you think upgrading it to the latest working **major** will be fine? I didn't manage to run the tests here.

**Comment by kibertoad** [comment]
@micalevisk Depends on what Node version they support. What were the breaking changes for 1.0.0?

**Comment by micalevisk** [comment]
well that will be hard to tell as there's no changelog to look at https://github.com/Marak/colors.js/compare/v0.6.2...v1.0.0

Keeping the current semver range won't cover the latest version of `colors`.

I didn't really get how `colors` is being used by `forever` tbh. I've just found this one https://github.com/foreversd/forever/blob/2211e32a288b97c3c3d1e27f41370a9a489ee833/lib/forever/cli.js#L629

**Comment by kibertoad** [comment]
Probably we need to replace it with colorette :)

**Comment by jerome-yvan** [comment]
Hi, I think the problem is on prettyjson module. It uses the colors.js version 1.4.2. Anyway on how to fix it?

**Comment by micalevisk** [comment]
oh, yeah

![image](https://user-images.githubusercontent.com/13461315/148706311-578cf9a4-455f-4255-85de-deac995741a6.png)

I guess we only need to wait them https://github.com/rafeca/prettyjson/pull/54

**Comment by iplanwebsites** [comment]
Forever crashing is causing many apps & servers to be offline right now. It'd be great if we could switch to a prettyJson fork temporarily if that PR can't make it shortly.

**Comment by lofairy** [comment]
```javascript
rm -rf /usr/lib/node_modules/forever/node_modules/prettyjson/node_modules/colors/
cd /usr/lib/node_modules/forever/node_modules/prettyjson
npm install colors@1.4.0
```

You can do this as an temporary solution before `prettyjson` apply rafeca/prettyjson#54.

**Comment by lofairy** [comment]
https://github.com/Marak/colors.js/commit/074a0f8ed0c31c35d13d28632bd8a049ff136fb6

WTF?

**Comment by iplanwebsites** [comment]
Had some servers using `npx forever` in built environments that were particularly complex to patch.

**Comment by kibertoad** [comment]
there is a fixed version of prettyjson coming up, will release new forever when that happens

**Comment by kibertoad** [comment]
Fix released in 4.0.2

**Comment by jerome-yvan** [comment]
> Fix released in 4.0.2

Thank you

**Comment by iplanwebsites** [comment]
Thanks everyone for that patch!
